### PR TITLE
#87: Add credential-gated live Linear smoke tests

### LINKED ISSUE #87: Add credential-gated live Linear smoke tests
## Issue Setup

- UAT Required: No
- Related Parent Issue or Context: Dogfood readiness gap for Linear credential-gated smoke coverage

## Issue Goal

Add skipped-by-default credential-gated live Linear smoke tests for Jade Symphony's normalized Linear tracker adapter.

## Why Now

Jade Symphony has a Linear tracker adapter behind the normalized tracker boundary, but dogfood readiness still calls out missing credential-gated Linear smoke coverage. A read-only opt-in smoke is the safest next slice before any Linear mutation smoke tests.

## Issue Context

Local `cargo test` must stay deterministic and credential-free. Live Linear smoke coverage should only run when an explicit environment flag is set. This first slice should validate read readiness only and avoid creating, updating, or commenting on Linear issues.

### Decisions

- Use `JADE_LIVE_LINEAR_SMOKE=1` as the explicit opt-in flag.
- Keep this first slice read-only.

### Assumptions

- `LINEAR_API_KEY` plus the existing Linear fixture workflow shape is enough to build a live read workflow for operator-supplied project settings.

## Non-Negotiable Guardrails

- Do not mutate Linear state in these tests.
- Do not read or log secrets.
- Do not make default `cargo test` depend on network or credentials.
- Do not edit files under `docs/bootstrap/references/openai-symphony`.

### In Scope

- Add integration tests in a new or existing integration-test directory that skip unless `JADE_LIVE_LINEAR_SMOKE=1` is set.
- Require `LINEAR_API_KEY` only when the live smoke flag is set.
- Exercise a read-only Jade Symphony command against a Linear workflow.
- Add a short docs page explaining how operators can run the live Linear smoke.

### Out of Scope

- Live Linear mutation tests.
- GitHub smoke test changes.
- CI configuration changes.
- Dynamic `linear_graphql` app-server tool implementation.

### Target Repository / Package

- `Alive24/jade-symphony`

### Relevant Knowledge Sources

- `docs/dogfood-readiness.md`
- `examples/linear-fixture-workflow.md`
- `docs/bootstrap/references/openai-symphony/SPEC.md`
- `docs/bootstrap/references/openai-symphony/elixir/lib/symphony_elixir/linear/client.ex`

### Relevant Code Paths

- `src/tracker.rs`
- `src/config.rs`

## Current State

Linear fixture planning and live adapter code exist, but there is no skipped-by-default live Linear smoke test harness.

## Deliverable Shape

Skipped-by-default integration tests plus a concise operator runbook.

## Risks or Constraints

- Live tests must be safe to run locally without altering tracker state.
- Test output must not include tokens or secrets.
- Missing credentials should produce a skipped test unless the opt-in flag is set.

## Expected Outcome

Operators can intentionally run a live Linear read smoke with `JADE_LIVE_LINEAR_SMOKE=1` while ordinary `cargo test` remains fast and credential-free.

### Completion Criteria

- Default `cargo test` passes with live Linear smoke skipped.
- Opt-in behavior is documented.
- The tests invoke the real binary instead of duplicating CLI logic.

### Functional Verification

- `cargo test`
- `cargo fmt --check`
- `cargo clippy --all-targets --all-features -- -D warnings`

### UAT

- Not required.

### Context Verification

- Confirm the tests are read-only and do not require Linear writes.

### PR DESCRIPTION
**Pull request by Alive24**
## Summary

- Add skipped-by-default live Linear smoke integration test.
- Gate live execution behind `JADE_LIVE_LINEAR_SMOKE=1`.
- Build a temporary live Linear workflow that references `$LINEAR_API_KEY` without writing the secret.
- Add a short runbook for operators.

## Verification

- `cargo test`
- `cargo fmt --check`
- `cargo clippy --all-targets --all-features -- -D warnings`

Closes #87

### CODE REVIEW

**Review by Alive24 — COMMENTED** [review]
## 📋 Review Summary
Skipped due to error: Command '['gemini', '-p', '\nYou are a Code Review Architect. Follow these guidelines strictly:\n---\nname: code-review-commons\ndescription: Common guidelines, persona and critical constraints for performing high-quality code reviews. Use this skill when performing a /code-review or /pr-code-review command.\nuser-invokable: false\n---\n\n# Code Review Commons\n\n## PERSONA\n\nYou are a very experienced **Principal Software Engineer** and a meticulous **Code Review Architect**. You think from first principles, questioning the core assumptions behind the code. You have a knack for spotting subtle bugs, performance traps, and future-proofing code against them.\n\n## OBJECTIVE\n\nYour task is to deeply understand the **intent and context** of the provided code changes (diff content) and then perform a **thorough, actionable, and objective** review.\nYour primary goal is to **identify potential bugs, security vulnerabilities, performance bottlenecks, and clarity issues**.\nProvide **insightful feedback** and **concrete, ready-to-use code suggestions** to maintain high code quality and best practices. Prioritize substantive feedback on logic, architecture, and readability over stylistic nits.\n\n## Instructions\n\n1. **Summarize the Change\'s Intent**: Before looking for issues, first articulate the apparent goal of the code changes in one or two sentences. Use this understanding to frame your review.\n2. **Establish context** by reading relevant files. Prioritize:\n    a. All files present in the diff.\n    b. Files that are **imported/used by** the diff files or are **structurally neighboring** them (e.g., related configuration or test files).\n3. **Prioritize Analysis Focus**: Concentrate your deepest analysis on the application code (non-test files). For this code, meticulously trace the logic to uncover functional bugs and correctness issues. Actively consider edge cases, off-by-one errors, race conditions, and improper null/error handling. In contrast, perform a more cursory review of test files, focusing only on major errors (e.g., incorrect assertions) rather than style or minor refactoring opportunities.\n4. **Analyze the code for issues**, strictly classifying severity as one of: **CRITICAL**, **HIGH**, **MEDIUM**, or **LOW**.\n\n## Critical Constraints\n\n**STRICTLY follow these rules for review comments:**\n\n* **Location:** You **MUST** only provide comments on lines that represent actual changes in the diff. This means your comments must refer **only to lines beginning with `+` or `-`**. **DO NOT** comment on context lines (lines starting with a space).\n* **Relevance:** You **MUST** only add a review comment if there is a demonstrable **BUG**, **ISSUE**, or a significant **OPPORTUNITY FOR IMPROVEMENT** in the code changes.\n* **Tone/Content:** **DO NOT** add comments that:\n    * Tell the user to "check," "confirm," "verify," or "ensure" something.\n    * Explain what the code change does or validate its purpose.\n    * Explain the code to the author (they are assumed to know their own code).\n    * Comment on missing trailing newlines or other purely stylistic issues that do not affect code execution or readability in a meaningful way.\n* **Substance First:** **ALWAYS** prioritize your analysis on the **correctness** of the logic, the **efficiency** of the implementation, and the **long-term maintainability** of the code.\n* **Technical Detail:**\n    * Pay **meticulous attention to line numbers and indentation** in code suggestions; they **must** be correct and match the surrounding code.\n    * **NEVER** comment on license headers, copyright headers, or anything related to future dates/versions (e.g., "this date is in the future").\n* **Formatting/Structure:**\n    * Keep the **change summary** concise (aim for a single sentence).\n    * Keep **comment bodies concise** and focused on a single issue.\n    * If a similar issue exists in **multiple locations**, state it once and indicate the other locations instead of repeating the full comment.\n    * **AVOID** mentioning your instructions, settings, or criteria in the final output.\n\n**Severity Guidelines (for consistent classification):**\n\n* **Functional correctness bugs that lead to behavior contrary to the change\'s intent should generally be classified as HIGH or CRITICAL.**\n* **CRITICAL:** Security vulnerabilities, system-breaking bugs, complete logic failure.\n* **HIGH:** Performance bottlenecks (e.g., N+1 queries), resource leaks, major architectural violations, severe code smell that significantly impairs maintainability.\n* **MEDIUM:** Typographical errors in code (not comments), missing input validation, complex logic that could be simplified, non-compliant style guide issues (e.g., wrong naming convention).\n* **LOW:** Refactoring hardcoded values to constants, minor log message enhancements, comments on docstring/Javadoc expansion, typos in documentation (.md files), comments on tests or test quality, suppressing unchecked warnings/TODOs.\n\n\nReview this PR.\nTitle: #87: Add credential-gated live Linear smoke tests\nBody: ## Summary\n\n- Add skipped-by-default live Linear smoke integration test.\n- Gate live execution behind `JADE_LIVE_LINEAR_SMOKE=1`.\n- Build a temporary live Linear workflow that references `$LINEAR_API_KEY` without writing the secret.\n- Add a short runbook for operators.\n\n## Verification\n\n- `cargo test`\n- `cargo fmt --check`\n- `cargo clippy --all-targets --all-features -- -D warnings`\n\nCloses #87\n\n\nDiff:\n```diff\ndiff --git a/docs/live-linear-smoke-tests.md b/docs/live-linear-smoke-tests.md\nnew file mode 100644\nindex 0000000..e124187\n--- /dev/null\n+++ b/docs/live-linear-smoke-tests.md\n@@ -0,0 +1,40 @@\n+# Live Linear Smoke Tests\n+\n+These integration tests are opt-in checks for the normalized Linear tracker\n+adapter. They are read-only in this slice, and ordinary `cargo test` skips them\n+without credentials.\n+\n+## Prerequisites\n+\n+- `curl` is installed.\n+- `LINEAR_API_KEY` is set.\n+- `JADE_LINEAR_PROJECT_SLUG` names a Linear project visible to that token.\n+\n+No tokens or secrets are written to the generated workflow file. The workflow\n+uses `$LINEAR_API_KEY`, which Jade resolves at runtime.\n+\n+## Run\n+\n+```bash\n+JADE_LIVE_LINEAR_SMOKE=1 \\\n+LINEAR_API_KEY=... \\\n+JADE_LINEAR_PROJECT_SLUG=your-project-slug \\\n+cargo test --test live_linear_smoke\n+```\n+\n+The smoke runs:\n+\n+- `jade-symphony inspect <temporary-linear-live-workflow.md>`\n+\n+## Expected Behavior\n+\n+- Without `JADE_LIVE_LINEAR_SMOKE=1`, the test prints a skip message and passes.\n+- With the flag set, missing `LINEAR_API_KEY` or `JADE_LINEAR_PROJECT_SLUG`\n+  fails immediately.\n+- The smoke must not create, update, comment on, or transition Linear issues.\n+\n+## Follow-Ups\n+\n+Mutation smoke tests for state updates, workpad comments, follow-up creation,\n+and project assignment should remain separate opt-in slices with stronger\n+guardrails.\ndiff --git a/tests/live_linear_smoke.rs b/tests/live_linear_smoke.rs\nnew file mode 100644\nindex 0000000..9ebdefe\n--- /dev/null\n+++ b/tests/live_linear_smoke.rs\n@@ -0,0 +1,92 @@\n+use std::fs;\n+use std::path::PathBuf;\n+use std::process::{Command, Output};\n+\n+use tempfile::TempDir;\n+\n+const LIVE_SMOKE_ENV: &str = "JADE_LIVE_LINEAR_SMOKE";\n+const PROJECT_SLUG_ENV: &str = "JADE_LINEAR_PROJECT_SLUG";\n+\n+fn live_smoke_enabled() -> bool {\n+    matches!(\n+        std::env::var(LIVE_SMOKE_ENV).as_deref(),\n+        Ok("1") | Ok("true") | Ok("yes")\n+    )\n+}\n+\n+fn repo_root() -> &\'static str {\n+    env!("CARGO_MANIFEST_DIR")\n+}\n+\n+fn combined_output(output: &Output) -> String {\n+    format!(\n+        "stdout:\\n{}\\nstderr:\\n{}",\n+        String::from_utf8_lossy(&output.stdout),\n+        String::from_utf8_lossy(&output.stderr)\n+    )\n+}\n+\n+fn require_live_smoke() -> Option<String> {\n+    if !live_smoke_enabled() {\n+        eprintln!("skipping live Linear smoke; set {LIVE_SMOKE_ENV}=1 to enable");\n+        return None;\n+    }\n+\n+    assert!(\n+        std::env::var_os("LINEAR_API_KEY").is_some(),\n+        "live Linear smoke requires LINEAR_API_KEY"\n+    );\n+\n+    let project_slug = std::env::var(PROJECT_SLUG_ENV)\n+        .unwrap_or_else(|_| panic!("live Linear smoke requires {PROJECT_SLUG_ENV}"));\n+    Some(project_slug)\n+}\n+\n+fn live_workflow(project_slug: &str) -> (TempDir, PathBuf) {\n+    let tempdir = tempfile::tempdir().expect("failed to create tempdir");\n+    let workflow_path = tempdir.path().join("linear-live-workflow.md");\n+    let workspace_root = tempdir.path().join("workspaces");\n+    let logs_root = tempdir.path().join("logs");\n+    let workflow = format!(\n+        r#"---\n+tracker:\n+  kind: linear\n+  project_slug: {project_slug:?}\n+  api_key: $LINEAR_API_KEY\n+workspace:\n+  root: {workspace_root:?}\n+observability:\n+  logs_root: {logs_root:?}\n+---\n+Live Linear read smoke.\n+"#\n+    );\n+    fs::write(&workflow_path, workflow).expect("failed to write live workflow");\n+    (tempdir, workflow_path)\n+}\n+\n+fn run_jade(args: &[&str]) -> Output {\n+    Command::new(env!("CARGO_BIN_EXE_jade-symphony"))\n+        .args(args)\n+        .current_dir(repo_root())\n+        .output()\n+        .expect("failed to execute jade-symphony binary")\n+}\n+\n+#[test]\n+fn live_linear_project_inspect_smoke() {\n+    let Some(project_slug) = require_live_smoke() else {\n+        return;\n+    };\n+    let (_tempdir, workflow_path) = live_workflow(&project_slug);\n+    let workflow_arg = workflow_path.to_string_lossy().to_string();\n+\n+    let output = run_jade(&["inspect", &workflow_arg]);\n+    let text = combined_output(&output);\n+    assert!(output.status.success(), "{text}");\n+    assert!(text.contains("issues="), "{text}");\n+    assert!(\n+        !text.contains("missing Linear API token"),\n+        "inspect should not report missing Linear auth when live smoke is enabled\\n{text}"\n+    );\n+}\n\n```\n\nWrite a review markdown file exactly in this format:\n## 📋 Review Summary\n[2-3 sentences summarizing the PR]\n## 🔍 General Feedback\n- [Feedback point 1]\n- [Feedback point 2]\n<COMMENT>\n[SEVERITY] [COMMENT_TEXT]\n</COMMENT>\n\n(Only add <COMMENT> if there\'s inline feedback). Do not include any other text before or after.\n']' timed out after 60 seconds
## 🔍 General Feedback
- None

**Review by Alive24 — COMMENTED** [review]
## 📋 Review Summary

This PR adds credential-gated live smoke tests for Linear integrations, matching the verification rigor applied to the GitHub adapter.

## 🔍 General Feedback

- **Parity in Testing**: Expanding live smoke testing to the Linear adapter ensures both primary tracker implementations are held to the same confidence standard.
- **Safe Execution**: Like the GitHub tests, gating these behind credentials prevents offline or unauthenticated test failures.
